cnd/src/network/comit.rs

```diff
@@ -254,6 +210,49 @@ impl Comit {
         // We trust in libp2p to poll us.
         Poll::Pending
     }
+
+    fn remote_data_insert<T>(&mut self, shared_swap_id: SharedSwapId, value: T)
+    where
+        RemoteData: Set<T>,
+    {
+        let mut remote_data = self
+            .remote_data
+            .get(&shared_swap_id)
+            .cloned()
+            .unwrap_or_default();
+        remote_data.set(value);
+        self.remote_data.insert(shared_swap_id, remote_data);
+    }
+
+    fn finalize(&mut self, peer: PeerId, shared_swap_id: SharedSwapId) {
+        let local_swap_id = match self.swaps.get_local_swap_id(shared_swap_id) {
+            Some(id) => id,
+            None => return,
+        };
+
+        let state = self.communication_states.get(&shared_swap_id);
+        let data = self.swaps.get_local_data(&local_swap_id);
+        let remote_data = self.remote_data.get(&shared_swap_id);
+
+        if let (Some(state), Some(data), Some(remote_data)) = (state, data, remote_data) {
+            let ethereum_sorted = if data.ethereum_identity.is_some() {
```

Comment:
❓ I don't really understand the `_sorted` naming - what is sorted and sorted by what?

Reply:
This "sorted" means "it is done", like "We've got this sorted" :)

Doesn't have anything to do with sorting things in a list.

Reply:
I balked at this name also, this justifies a ticket IMO (or even an inline `TODO` ;)

Reply:
I'd propose `_received`, because I think that is the point - that we received information (identities, secret-hash...) from the other party.
Happy to do it in a follow up, if you tackle it let me know.